### PR TITLE
6 game physics

### PR DESCRIPTION
**Pull request by ju-lit-hein**

### CODE REVIEW

File: client/include/ECS/Components/BoxComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/BoxComponent.hpp:1]: Document that

File: client/include/ECS/Components/HitBoxComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/HitBoxComponent.hpp:1]: Same

File: client/include/ECS/Components/HurtBoxComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/HurtBoxComponent.hpp:1]: Same

File: client/include/ECS/Components/ModelComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/ModelComponent.hpp:1]: Same

File: client/include/ECS/Components/MotionComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/MotionComponent.hpp:1]: Same

File: client/include/ECS/Components/SpriteComponent.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Components/SpriteComponent.hpp:1]: Same

File: client/include/ECS/Systems/Render2DSystem.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Systems/Render2DSystem.hpp:1]: Same

File: client/include/ECS/Systems/Render3DSystem.hpp
Comment (MAILLOTNathan) [inline client/include/ECS/Systems/Render3DSystem.hpp:1]: Same

File: include/ECS/Components/HealthComponent.hpp
Comment (MAILLOTNathan) [inline include/ECS/Components/HealthComponent.hpp:1]: Same

File: include/ECS/Components/TransformComponent.hpp
Comment (MAILLOTNathan) [inline include/ECS/Components/TransformComponent.hpp:1]: Same

**Review by MAILLOTNathan — CHANGES_REQUESTED** [review]

File: client/include/ECS/Components/BoxComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/BoxComponent.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Components/HitBoxComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/HitBoxComponent.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Components/HurtBoxComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/HurtBoxComponent.hpp:1]: I documented it @MAILLOTNathan

File: include/ECS/Components/TransformComponent.hpp
Comment (ju-lit-hein) [reply include/ECS/Components/TransformComponent.hpp:1]: I documented it @MAILLOTNathan

File: include/ECS/Components/HealthComponent.hpp
Comment (ju-lit-hein) [reply include/ECS/Components/HealthComponent.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Systems/Render3DSystem.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Systems/Render3DSystem.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Systems/Render2DSystem.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Systems/Render2DSystem.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Components/SpriteComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/SpriteComponent.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Components/MotionComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/MotionComponent.hpp:1]: I documented it @MAILLOTNathan

File: client/include/ECS/Components/ModelComponent.hpp
Comment (ju-lit-hein) [reply client/include/ECS/Components/ModelComponent.hpp:1]: I documented it @MAILLOTNathan

**Comment by ju-lit-hein** [comment]
Please do not close the branch after merging it into main @MAILLOTNathan

**Pull request merged**